pricegraph/src/graph/path.rs

```diff
@@ -34,6 +52,21 @@ impl<N: PartialEq> NegativeCycle<N> {
             }
         }
     }
+
+    /// Returns an iterator over the nodes in the cycle in the order
+    /// they appear.
+    pub fn iter(&self) -> Iter<N> {
```

Comment:
I was thinking about the [iter trait](https://doc.rust-lang.org/std/iter/index.html), but that works too.

Reply:
Implementing `Iterator` for `NegativeCycle` means that the cycle itself can be iterated, so for every call to `.next()` the cycle would become smaller. This seems strange to me, while having something in the cycle that returns an iterator would be more intuitive.